src/straPy/data_vis.py

```diff
@@ -1,58 +1,140 @@
+    if not isinstance(title, str):
+        raise TypeError("The value of the argument 'title' must be type of str.")
+        
+    if not isinstance(x_axis, str):
+        raise TypeError("The value of the argument 'x_axis' must be type of str.")
+        
+    if not isinstance(y_axis, str):
+        raise TypeError("The value of the argument 'y_axis' must be type of str.")
+        
+    plt.hist(calculate_boot_stats(sample, rep, level=ci_level, random_seed = ci_random_seed, pass_dist=True)[1], density=False, bins=bin_size)
+    plt.axvline(calculate_boot_stats(sample, rep, level=ci_level, random_seed = ci_random_seed, pass_dist=True)[0]["lower"], color='k', linestyle='--')
+    plt.axvline(calculate_boot_stats(sample, rep, level=ci_level, random_seed = ci_random_seed, pass_dist=True)[0]["sample_mean"], color='r', linestyle='-')
+    plt.axvline(calculate_boot_stats(sample, rep, level=ci_level, random_seed = ci_random_seed, pass_dist=True)[0]["upper"], color='k', linestyle='--')
+    plt.title(title)
+    plt.xlabel(x_axis)
+    plt.ylabel(y_axis)
 
-    A table that contains the provided sampling distribution`dist`'s 
-    confidence interval `ci` and other relevant statistics, including
-    standard error `se`, sample size `size`, sample mean `mean` and 
-    selected alpha `alpha`.
+
+def summary_tables(stat_dic, precision=2, estimator=True, alpha=True):
+    """Makes two tables that summerize the statistics from the bootstrapped samples and the parameters for creating the bootstrapped samples.
 
 
     Parameters
     ----------
-    dist : numpy.ndarray
-        bootstrapped sampling distribution
-    ci : numpy.ndarray
-        confidence interval
-    size : boolean, default=True
-        sample size
-    mean: boolean, default=True
-        sample mean
-    alpha : float, default=0.05
-        selected alpha
+    stat_dic : dict
+        summary statistics produced by the `calculate_boot_stats()` function 
+    precision : int, default= 2
+        the precision of the table values
+    esitmator : boolean, default=True
+        include the bootstrap estimate in the summary statistics table
+    alpha: boolean, default=True
+        include the significance level in the summary statistics table
 
     Returns
     -------
-    table object
-        table with lower bound and upper bound of the confidence interval and relevant statistics
+    summary statistics: table object
+        table summerizing the lower bound and upper bound of the confidence interval, the standard error, the sampling statitic (if estimator = True), and the significance level (if alpha = True)
+    bootstrap parameters: table object
+        table  summerizing the parameters of the bootstrap sampling spficiying the original sample size, number of repititions, the significance level, and the number of samples in each bootstrap if its different from the original sample size.  
```

Comment:
According to Python's pep8 styling guide, the number of characters should not be more than 79 characters. Can you break these lines?

Reply:
This has been updated